src/wp-includes/wp-db.php

```diff
@@ -1347,6 +1367,35 @@ public function escape_by_ref( &$string ) {
 		}
 	}
 
+	/**
+	 * Escapes an identifier for a MySQL database (e.g. table/field names).
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param string $identifier Identifier to escape.
+	 */
+	public function escape_identifier( $identifier ) {
+		return '`' . $this->_escape_identifier_value( $identifier ) . '`';
+	}
+
+	/**
+	 * Escapes an identifier value.
+	 *
+	 * Escapes an identifier value without adding the surrounding quotes.
+	 *
+	 * - Permitted characters in quoted identifiers include the full Unicode Basic Multilingual Plane (BMP), except U+0000
+	 * - To quote the identifier itself, then you need to double the character, e.g. `a``b`
+	 *
+	 * @link https://dev.mysql.com/doc/refman/8.0/en/identifiers.html
+	 * @since 6.0.0
+	 * @access private
+	 *
+	 * @param string $identifier Identifier to escape.
+	 */
+	private function _escape_identifier_value( $identifier ) {
+		return str_replace( '`', '``', $identifier );
```

Comment:
Do we ever need to account for `sql_mode='ANSI_QUOTES'`?

Reply:
Fortunately this approach works for `ANSI_QUOTES ` as well. WordPress uses MySQL/MariaDB, where "You can still use ``` ` ``` to quote identifiers with this mode enabled" ([source](https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sqlmode_ansi_quotes)), and double back-ticks are still used for escaping.